apis/v1alpha2/gateway_types.go

```diff
@@ -518,12 +518,20 @@ const (
 	// Support: Extended
 	IPAddressType AddressType = "IPAddress"
 
+	// A hostname represents a DNS based ingress point. This is similar to the
+	// corresponding hostname field in Kubernetes load balancer status. For
+	// example, this concept may be used for cloud load balancers where a DNS
+	// name is used to expose a load balancer.
+	//
+	// Support: Extended
+	HostnameAddressType AddressType = "Hostname"
+
 	// An opaque identifier that represents a specific IP address. The
-	// interpretation of the name is dependent on the controller. For
-	// example, a "NamedAddress" might be a cloud-dependent identifier
-	// for a static or elastic IP.
+	// interpretation of the name is dependent on the controller. For example, a
+	// "NamedAddress" might be a name or other cloud-specific identifier that
+	// refers to a static or elastic IP on a cloud provider.
```

Comment:
Should we remove this line?
Reading up to `cloud-specific identifier` makes sense but as soon as I read `static or elastic IP`, I get confused with IPAddressType. I know that `refers to` is the important part but I think we should come up with some better examples or skip examples here. Is this type intended to be used for something like [AWS ARNs](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html)?

Reply:
Yeah I'm not really sure how to describe this well. I think an ARN is a great example of how a reference could work, but I'm hesitant to add too many specific examples here. IE if we suggest that an AWS ARN could be used here, users may assume that it works with whatever implementation they're using which may not be the case. 

At the same time, I know we did not have the `static or elastic IP` example initially but that resulted in confusion as far as what this should be used to refer to. I agree that there's room for improvement here but haven't found a great way to do that yet without adding more confusion.

Reply:
Took another shot at this, but not sure it's any better. PTAL.